packages/react/src/metadata.tsx

```diff
@@ -414,7 +414,11 @@ const isNotRelatedToSpecialModelFilter = (field: FieldMetadata) => {
 
 const specialModelKeys = new Set(["DataModel-Shopify-Shop"]);
 
-const isAcceptedFieldTypeFilter = (field: FieldMetadata) => acceptedAutoFormFieldTypes.has(field.fieldType);
+const isAcceptedFieldTypeFilter = (allowIdFieldType?: boolean) => {
```

Comment:
Again, I would prefer to avoid behavioral flag parameters. You could achieve the same effect by introducing a new function like

```
const isAcceptedUpsertFieldTypeFilter = (field: FieldMetadata) => isAcceptedFieldTypeFilter(field.fieldType) || field.fieldType === FieldType.Id);

Reply:
Updated such that the filters for upsert/nonUpsert are different and selected via ternary